### PR TITLE
#1073 twilio sms translate

### PR DESCRIPTION
**Pull request by jakehova**
# Description

#1073 - add delivery status translation to notify platform delivery status

# (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Ran tests against valid trigger events
- [x] Ran tests against celery message creation
- [x] Ran tests with various status codes from Twilio
- [x] Ran tests with various error codes from Twilio

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/clients/sms/twilio.py [hunk 1]
```diff
@@ -110,12 +151,55 @@ def send_sms(self, to, content, reference, **kwargs) -> str:
 
                 self.logger.info(f"Twilio message created using messaging_service_sid")
 
-            self.logger.info(f"Twilio send SMS request for {reference} succeeded: {message.sid}")
+            self.logger.info(
+                f"Twilio send SMS request for {reference} succeeded: {message.sid}"
```
Comment (k-macmillan) [inline app/clients/sms/twilio.py:155]: No f-strings with loggers.

File: app/clients/sms/twilio.py [hunk 2]
```diff
@@ -110,12 +151,55 @@ def send_sms(self, to, content, reference, **kwargs) -> str:
...
 
             return message.sid
         except Exception as e:
             self.logger.error(f"Twilio send SMS request for {reference} failed")
             raise e
         finally:
             elapsed_time = monotonic() - start_time
-            self.logger.info(f"Twilio send SMS request for {reference} finished in {elapsed_time}")
+            self.logger.info(
+                f"Twilio send SMS request for {reference} finished in {elapsed_time}"
```
Comment (k-macmillan) [inline app/clients/sms/twilio.py:165]: no f-strings with loggers.

File: app/clients/sms/twilio.py [hunk 3]
```diff
@@ -110,12 +151,55 @@ def send_sms(self, to, content, reference, **kwargs) -> str:
...
+            )
+
+    def translate_delivery_status(self, twilio_delivery_status_message) -> dict:
```
Comment (k-macmillan) [inline app/clients/sms/twilio.py:168]: Missing docstring as well as annotations and logging.

**Review by k-macmillan — CHANGES_REQUESTED** [review]

File: app/clients/sms/twilio.py [hunk 4]
```diff
@@ -110,12 +151,66 @@ def send_sms(self, to, content, reference, **kwargs) -> str:
...
+        ):
+            error_code = parsed_dict["ErrorCode"][0]
+
+            if error_code in twilio_error_code_map:
+                notify_delivery_status = twilio_error_code_map[error_code]
+            else:
+                notify_delivery_status = twilio_notify_status_map[
+                    twilio_delivery_status
+                ]
+        else:
```
Comment (k-macmillan) [inline app/clients/sms/twilio.py:205]: This does not cover a cancel status. Also, if any of the other ones throw an error code but are not covered it'd come here. Under what circumstances will Twilio have an error code but be anything but failed, undelivered, or canceled?

**Review by k-macmillan — CHANGES_REQUESTED** [review]

File: app/clients/sms/twilio.py (hunk 4 above)
Comment (jakehova) [reply app/clients/sms/twilio.py:205]: error codes only occur for failed and undelivered.  From Twilio Docs:
```
When a message's status is failed or undelivered, the ErrorCode and ErrorMessage properties will contain one of the following.
```

File: requirements-app.txt [hunk 5]
```diff
@@ -15,7 +15,7 @@ celery[sqs]>=4.4.7
 click-datetime>=0.2
 
 # Feb 2022: Package for Datadog Tracer
-ddtrace==1.8.0
+# ddtrace==1.8.0
```
Comment (k-macmillan) [inline requirements-app.txt:18]:
```suggestion
ddtrace==1.8.0
```

File: application.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 from __future__ import print_function
-from ddtrace import patch_all
+# from ddtrace import patch_all
```
Comment (k-macmillan) [inline application.py:3]:
```suggestion
from ddtrace import patch_all
```

File: application.py [hunk 7]
```diff
@@ -27,4 +27,4 @@
 
 
 # this starts the ddtrace tracer and configures it to the right port and URL
-patch_all()
+# patch_all()
```
Comment (k-macmillan) [inline application.py:30]:
```suggestion
patch_all()
```

**Review by k-macmillan — CHANGES_REQUESTED** [review]

File: requirements-app.txt (hunk 5 above)
Comment (jakehova) [reply requirements-app.txt:18]: sorry, did this in baldwin's branch and not here

**Review by ldraney — APPROVED** [review]

**Review by k-macmillan — APPROVED** [review]

**Pull request merged**